cli/src/action/registry/mod.rs

```diff
@@ -172,3 +181,149 @@ impl Action for RegistryGenerateAction {
+
+        let identity = args
+            .value_of("identity")
+            .ok_or_else(|| CliError::ActionError("Identity must be specified".into()))?
+            .to_string();
+
+        let display_name = args
+            .value_of("display_name")
+            .unwrap_or(&identity)
+            .to_string();
+
+        let mut builder = SplinterRestClientBuilder::new();
+        builder = builder.with_url(url);
+
+        #[cfg(feature = "splinter-cli-jwt")]
+        {
+            let private_key = args.value_of("private_key_file");
+            builder = builder.with_auth(create_cylinder_jwt_auth(private_key)?);
+        }
+
+        let client = builder.build()?;
+
+        let mut node_metadata: HashMap<String, String> = HashMap::new();
+        if let Some(metadata) = args.values_of("metadata") {
+            for pair in metadata {
+                let (key, value) = parse_metadata(pair)?;
+                node_metadata.insert(key, value);
+            }
+        }
+
+        if args.is_present("from_remote") {
+            let remote_node = client.get_node(&identity)?.ok_or_else(|| {
+                CliError::ActionError("Unable to retrieve node from remote".into())
+            })?;
+
+            let node = RegistryNode {
+                identity: remote_node.identity,
+                endpoints: remote_node.endpoints,
+                display_name: remote_node.display_name,
+                keys: remote_node.keys,
+                metadata: remote_node.metadata,
+            };
+
+            if !args.is_present("dry_run") {
+                client.add_node(&node)?;
+            }
+
+            info!("{}", node);
+
+            Ok(())
+        } else {
+            let endpoints: Vec<String> = args
+                .values_of("endpoint")
+                .ok_or_else(|| {
+                    CliError::ActionError("One or more endpoints must be specified".into())
+                })?
+                .map(String::from)
+                .collect::<Vec<String>>();
+
+            let keys: Vec<String> = args
+                .values_of("key_files")
+                .ok_or_else(|| {
+                    CliError::ActionError("One or more key files must be specified".into())
+                })?
+                .map(|key_file| read_private_key(key_file))
+                .collect::<Result<_, _>>()?;
+
+            let node = RegistryNode {
+                identity,
+                endpoints,
+                display_name,
+                keys,
+                metadata: node_metadata,
+            };
+
+            if !args.is_present("dry_run") {
+                client.add_node(&node)?
+            }
+
+            info!("{}", node);
+
+            Ok(())
+        }
+    }
+}
+
+#[cfg(feature = "registry")]
+fn parse_metadata(metadata: &str) -> Result<(String, String), CliError> {
+    let mut parts = metadata.splitn(2, ':');
+    match (parts.next(), parts.next()) {
+        (Some(key), Some(value)) => match key {
```

Comment:
I belive `value` could be an empty string as well if the metadata arg is `key:`.

Reply:
added a match to check for that case